include/ibamr/LevelSetUtilities.h

```diff
@@ -0,0 +1,420 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (c) 2017 - 2020 by the IBAMR developers
+// All rights reserved.
+//
+// This file is part of IBAMR.
+//
+// IBAMR is free software and is distributed under the 3-clause BSD
+// license. The full text of the license can be found in the file
+// COPYRIGHT at the top level directory of IBAMR.
+//
+// ---------------------------------------------------------------------
+
+/////////////////////////////// INCLUDE GUARD ////////////////////////////////
+
+#ifndef included_IBAMR_LevelSetUtilities
+#define included_IBAMR_LevelSetUtilities
+
+/////////////////////////////// INCLUDES /////////////////////////////////////
+
+#include <ibamr/config.h>
+
+#include "ibamr/LSInitStrategy.h"
+
+#include "tbox/DescribedClass.h"
+#include "tbox/Pointer.h"
+#include "tbox/Serializable.h"
+
+#include <limits>
+#include <string>
+
+namespace SAMRAI
+{
+namespace hier
+{
+template <int DIM>
+class PatchHierarchy;
+}
+namespace pdat
+{
+template <int DIM, class TYPE>
+class CellVariable;
+} // namespace pdat
+namespace tbox
+{
+class Database;
+} // namespace tbox
+} // namespace SAMRAI
+
+namespace IBAMR
+{
+class AdvDiffHierarchyIntegrator;
+} // namespace IBAMR
+
+namespace IBTK
+{
+class HierarchyMathOps;
+}
+
+/////////////////////////////// CLASS DEFINITION /////////////////////////////
+
+namespace IBAMR
+{
+/*!
+ * \brief Class LevelSetUtilities provides implementation of some helper functions
+ * to enforce mass/volume conservation of phases.
+ */
+
+class LevelSetUtilities
+{
+public:
+    /*!
+     * \brief A lightweight class to hold the level set variable and the associated hierarchy
+     * integrator (AdvDiffHierarchyIntegrator).
+     */
+    class LevelSetContainer : public virtual SAMRAI::tbox::DescribedClass
+    {
+    public:
+        /*!
+         * \brief Constructor of the class.
+         */
+        LevelSetContainer(SAMRAI::tbox::Pointer<AdvDiffHierarchyIntegrator> adv_diff_integrator,
+                          SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > ls_var)
+            : d_adv_diff_integrator(adv_diff_integrator)
```

Comment:
I think this 
```suggestion
            : d_adv_diff_integrator(adv_diff_integrator),
              d_ls_vars({ls_var})
```

Reply:
I reverted this back, as this way of constructing vector data members does not work (at least on my system). 

I tried a simple program

```
#include <iostream>
#include <vector>

class A
{
public:

A(double a = 1.5, double b= 2.7):d_vec({a,b})
{
return;
}

void print()
{
  for (int i = 0; i < d_vec.size(); ++i)
  {
     std::cout << d_vec[i];
  }
  return;
}
private:
std::vector<double> d_vec;
};

int main()
{
   A obj;
   obj.print();
   return 0;
}

$ clang++ vect.cpp -o ./exe
vect.cpp:8:40: error: expected expression
A(double a = 1.5, double b= 2.7):d_vec({a,b})
                                                                   ^
1 error generated.


```